src/main/java/frc/robot/subsystems/Lift.java

```diff
@@ -192,11 +182,130 @@ public boolean rightLowerLimitReached() {
 
     @Override
     public void periodic() {
+        if (enabled) {
+            double output = verticalController.calculate(getLeftPosition());
+            double error = errorController.calculate(DriveTrain.getInstance().getPitch());
+
+            double power = MathUtil.clamp(output, LiftConstants.backwardsMotorSpeed, LiftConstants.motorSpeed);
+            double clampedError = MathUtil.clamp(error, LiftConstants.backwardsMotorSpeed, LiftConstants.motorSpeed);
+
+            double left = DriveTrain.getInstance().getPitch() > 0 ? power + clampedError : power;
+            double right = DriveTrain.getInstance().getPitch() < 0 ? power - clampedError : power;
+            runLeftUp(left);
+            runRightUp(right);
+        }
         if (leftLimitSwitch.isPressed() && leftLiftEncoder.getPosition() != 0) {
             leftLiftEncoder.setPosition(0);
         }
         if (rightLimitSwitch.isPressed() && rightLiftEncoder.getPosition() != 0) {
             rightLiftEncoder.setPosition(0);
         }
+
+    }
+
+    /**
+     * Returns the vertical PIDController.
+     *
+     * @return The vertical controller.
+     */
+    public PIDController getVerticalController() {
+        return verticalController;
+    }
+
+    /**
+     * Returns the Right PIDController.
+     *
+     * @return The right controller.
+     */
+    public PIDController getErrorController() {
+        return errorController;
+    }
+
+    /**
+     * Sets the Scoring PID setpoint to one of the constant setpoints.
+     *
+     * @param ElevatorSetpoints - desired setpoint
+     */
+    public void setSetpoint(double setpoint) {
+        verticalController.setSetpoint(MathUtil.clamp(setpoint, 0, LiftConstants.maxExtension));
+    }
+
+    /**
+     * Returns the current setpoint of the subsystem.
+     *
+     * @return The current setpoint
+     */
+    public double getSetpoint() {
+        return verticalController.getSetpoint();
+    }
+
+    /**
+     * Returns the position of the left lift encoder.
+     *
+     * @return double - current encoder position
```

Comment:
```suggestion
     * @return current encoder position
```
you don't need to mention the data type in javadoc, this is handled automatically